plugins/modules/sysrc.py

```diff
@@ -130,11 +131,12 @@ def exists(self):
         Use this dictionary to preform the tests.
         """
         (rc, out, err) = self.run_sysrc('-e', '-a')
-        conf = dict([i.split('=', 1) for i in out.splitlines()])
+        conf = dict((part.split('=', 1) for part in split(out, comments=True)))
```

Comment:
It would be interesting to add tests for the use case this PR fixes - couple of years from now we might forget completely the reason why we are doing this.